examples/experimental/dagster-airlift/dagster_airlift/in_airflow/mark_as_migrating.py

```diff
@@ -29,21 +32,61 @@ def mark_as_dagster_migrating(
     if not logger:
         logger = logging.getLogger("dagster_airlift")
     logger.debug(f"Searching for dags migrating to dagster{suffix}...")
-    num_dags = 0
-    for obj in global_vars.values():
+    migrating_dags = set()
+    # Do a pass to collect dags and ensure that migration information is set correctly.
+    for var, obj in global_vars.items():
         if not isinstance(obj, DAG):
             continue
         dag: DAG = obj
-        logger.debug(f"Checking dag with id `{dag.dag_id}` for migration state.")
-        migration_state_for_dag = migration_state.get_migration_dict_for_dag(dag.dag_id)
-        if migration_state_for_dag is None:
-            logger.debug(
-                f"Dag with id `{dag.dag_id} hasn't been marked with migration state. Skipping..."
+        if not migration_state.dag_is_being_migrated(dag.dag_id):
+            logger.debug(f"Dag with id `{dag.dag_id}` has no migration state. Skipping...")
+            continue
+        logger.debug(f"Dag with id `{dag.dag_id}` has migration state.")
+        migration_state_for_dag = migration_state.dags[dag.dag_id]
+        for task_id in migration_state_for_dag.tasks.keys():
+            if task_id not in dag.task_dict:
+                raise Exception(
+                    f"Task with id `{task_id}` not found in dag `{dag.dag_id}`. Found tasks: {list(dag.task_dict.keys())}"
+                )
+            if not isinstance(dag.task_dict[task_id], BaseOperator):
+                raise Exception(
+                    f"Task with id `{task_id}` in dag `{dag.dag_id}` is not an instance of BaseOperator. This likely means a MappedOperator was attempted, which is not yet supported by airlift."
+                )
+        migrating_dags.add(var)
+
+    for var in migrating_dags:
+        dag: DAG = global_vars[var]
```

Comment:
still think it is very confusing to use `var` here. I would just get the list of dags directly and not do the 2nd round of indirection through global_vars